### ISSUE TITLE
Fails using URI 1.60

### ISSUE THREAD

**Issue by faxm0dem** [plicease/AnyEvent-WebSocket-Client#20]
I'd love to post a PR but I'm not sure how as the Build.PL is missing on github

**Comment by plicease** [comment]
I downgraded to 1.60 and it works for me:

```
# perl                 v5.20.3
# AE                   7.11
# AnyEvent             7.11
# Devel::Cycle         1.12
# EV                   4.21
# FindBin              1.51
# IO::Socket::SSL      2.020
# Math::Random::Secure -
# Module::Build        0.4214
# Mojolicious          6.24
# Moo                  2.000002
# Net::SSLeay          1.72
# PerlX::Maybe         1.001
# PerlX::Maybe::XS     -
# Protocol::WebSocket  0.19
# Test::Memory::Cycle  1.04
# Test::More           1.001014
# Test::Warn           0.30
# URI                  1.60
# URI::ws              0.03
# 
# 
# 
t/00_diag.t .......................... ok   
t/ae.t ............................... ok   
t/ae_ssl.t ........................... ok   
t/client_masked.t .................... ok   
t/connection_after_close.t ........... ok   
t/connection_callback_send.t ......... ok   
t/connection_counter_close.t ......... ok   
t/connection_counter_shutdown.t ...... ok   
t/connection_destroy_in_callbacks.t .. ok   
t/connection_finish_callback.t ....... ok   
t/connection_huge_message.t .......... ok   
t/connection_masked.t ................ ok   
t/connection_ping_pong.t ............. ok   
t/echo.t ............................. ok   
t/message.t .......................... ok   
t/mojo.t ............................. ok   
t/mojo_close.t ....................... ok   
t/mojo_echo.t ........................ ok   
t/mojo_recieve.t ..................... ok   
t/mojo_scope.t ....................... ok   
t/mojo_ssl.t ......................... skipped: set ANYEVENT_WEBSOCKET_CLIENT_TEST_MOJO_SSL to enable this test
t/new.t .............................. ok   
t/server_initial_data.t .............. ok   
t/use.t .............................. ok   
```

If you can provide me some details as to exactly what is failing and any diagnostics produced, I'd be happy to take a look.

I use `Dist::Zilla` (which creates Build.PL) since it saves me time in development and release, but I know it  is hard for contributors who do not use that tool, so I am happy to accept patches to `Build.PL` from the CPAN version as a "place to start".  Patches or PRs that can be maid against `lib/My/ModuleBuild.pm` will probably be integrated much faster though.  I will likely need to be able to reproduce the problem though.

**Comment by faxm0dem** [comment]
Here's the test log, and sorry it's not 1.6 that is failing but URI 1.40:

```
# perl -MURI -le 'print $URI::VERSION'
1.40
# perl -MURI::ws -le 'print $URI::ws::VERSION'
0.03
AnyEvent-WebSocket-Client-0.31-0tZp_G]# ./Build  test
t/00_diag.t .......................... 1/1 #
#
#
# HARNESS_ACTIVE=1
# HARNESS_VERSION=3.17
# LANG=en_US.UTF-8
# PERL5LIB=/root/.cpan/build/AnyEvent-WebSocket-Client-0.31-0tZp_G/blib/lib:/root/.cpan/build/AnyEvent-WebSocket-Client-0.31-0tZp_G/blib/arch:inc:/opt/lib/perl5/x86_64-linux-thread-multi:/opt/lib/perl5:/opt/lib/perl5
# PERL_DL_NONLAZY=1
# SHELL=/bin/bash
#
#
#
# PERL5LIB path
# /root/.cpan/build/AnyEvent-WebSocket-Client-0.31-0tZp_G/blib/lib
# /root/.cpan/build/AnyEvent-WebSocket-Client-0.31-0tZp_G/blib/arch
# inc
# /opt/lib/perl5/x86_64-linux-thread-multi
# /opt/lib/perl5
# /opt/lib/perl5
#
#
#
# perl                 v5.10.1
# AE                   5.271
# AnyEvent             5.271
# Devel::Cycle         -
# EV                   -
# FindBin              1.50
# IO::Socket::SSL      -
# Math::Random::Secure -
# Module::Build        0.4208
# Mojolicious          -
# Moo                  2.000002
# Net::SSLeay          1.35
# PerlX::Maybe         1.001
# PerlX::Maybe::XS     -
# Protocol::WebSocket  0.18
# Test::Memory::Cycle  -
# Test::More           1.001003
# Test::Warn           -
# URI                  1.40
# URI::ws              0.03
#
#
#
t/00_diag.t .......................... ok
t/ae.t ............................... Can't locate object method "secure" via package "URI::ws" at /root/.cpan/build/AnyEvent-WebSocket-Client-0.31-0tZp_G/blib/lib/AnyEvent/WebSocket/Client.pm line 80.
# Looks like your test exited with 115 before it could output anything.
t/ae.t ............................... Dubious, test returned 115 (wstat 29440, 0x7300)
Failed 3/3 subtests
t/ae_ssl.t ........................... ok
t/client_masked.t .................... Can't locate object method "secure" via package "URI::ws" at /root/.cpan/build/AnyEvent-WebSocket-Client-0.31-0tZp_G/blib/lib/AnyEvent/WebSocket/Client.pm line 80.
t/client_masked.t .................... Dubious, test returned 115 (wstat 29440, 0x7300)
No subtests run
t/connection_after_close.t ........... ok
t/connection_callback_send.t ......... Can't locate object method "secure" via package "URI::ws" at /root/.cpan/build/AnyEvent-WebSocket-Client-0.31-0tZp_G/blib/lib/AnyEvent/WebSocket/Client.pm line 80.
# Looks like your test exited with 115 before it could output anything.
t/connection_callback_send.t ......... Dubious, test returned 115 (wstat 29440, 0x7300)
Failed 1/1 subtests
t/connection_counter_close.t ......... ok
t/connection_counter_shutdown.t ...... ok
t/connection_destroy_in_callbacks.t .. ok
t/connection_finish_callback.t ....... ok
t/connection_huge_message.t .......... ok
t/connection_masked.t ................ ok
t/connection_ping_pong.t ............. ok
t/echo.t ............................. Can't locate object method "secure" via package "URI::ws" at /root/.cpan/build/AnyEvent-WebSocket-Client-0.31-0tZp_G/blib/lib/AnyEvent/WebSocket/Client.pm line 80.
# Looks like your test exited with 115 before it could output anything.
t/echo.t ............................. Dubious, test returned 115 (wstat 29440, 0x7300)
Failed 4/4 subtests
t/message.t .......................... ok
t/mojo.t ............................. skipped: Requires EV
t/mojo_close.t ....................... skipped: Requires EV
t/mojo_echo.t ........................ skipped: Requires EV
t/mojo_recieve.t ..................... skipped: Requires EV
t/mojo_scope.t ....................... skipped: Requires EV
t/mojo_ssl.t ......................... skipped: Requires IO::Socket::SSL 1.75
t/new.t .............................. ok
t/server_initial_data.t .............. Can't locate object method "secure" via package "URI::ws" at /root/.cpan/build/AnyEvent-WebSocket-Client-0.31-0tZp_G/blib/lib/AnyEvent/WebSocket/Client.pm line 80.
t/server_initial_data.t .............. Dubious, test returned 115 (wstat 29440, 0x7300)
No subtests run
t/use.t .............................. ok

Test Summary Report
-------------------
t/ae.t                             (Wstat: 29440 Tests: 0 Failed: 0)
  Non-zero exit status: 115
  Parse errors: Bad plan.  You planned 3 tests but ran 0.
t/client_masked.t                  (Wstat: 29440 Tests: 0 Failed: 0)
  Non-zero exit status: 115
  Parse errors: No plan found in TAP output
t/connection_callback_send.t       (Wstat: 29440 Tests: 0 Failed: 0)
  Non-zero exit status: 115
  Parse errors: Bad plan.  You planned 1 tests but ran 0.
t/echo.t                           (Wstat: 29440 Tests: 0 Failed: 0)
  Non-zero exit status: 115
  Parse errors: Bad plan.  You planned 4 tests but ran 0.
t/server_initial_data.t            (Wstat: 29440 Tests: 0 Failed: 0)
  Non-zero exit status: 115
  Parse errors: No plan found in TAP output
Files=24, Tests=33,  5 wallclock secs ( 0.18 usr  0.17 sys +  3.53 cusr  0.90 csys =  4.78 CPU)
Result: FAIL
Failed 5/24 test programs. 0/33 subtests failed.
```

**Comment by plicease** [comment]
No problem.  Thanks for the report.  Looks like `secure` was added in 1.53, I should have an updated release soon.

**Comment by plicease** [comment]
I believe this to be fixed in 0.32.  Please reopen with details if you disagree.

Thanks again for the report.
